### PR TITLE
CIP5 - Common Bech32 Prefixes

### PR DESCRIPTION
**Pull request by KtorZ**
draft proposal for CIP-5

### CODE REVIEW

**Comment by SebastienGllmt** [comment]
You may want to reference https://github.com/satoshilabs/slips/blob/master/slip-0173.md

File: CIP5/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+Therefore, we can leverage bech32 for binary data encoding, with a set of common prefixes that can be used across tools and software to disambiguate payloads. 
+
+## Specification
+
+We define the following set of common prefixes with their corresponding semantic. Any software willing to represent binary data in a human-friendly way should abide by these guidelines. Should a data-type be missing, we encourage developers to update this CIP and register a new prefix. 
+
+| Prefix            | Semantic                                         |
+| ---               | ---                                              |
+| `addr`            | Mainnet address                                  |
+| `addr_test_{tag}` | A Testnet address with an associated network tag |
```
[the remark this answers is not in the document]
Comment (dcoutts) [reply CIP5/README.md:29]: Tentatively using `stake` and "stake_test` prefixes for stake addresses.

**Comment by KtorZ** [comment]
> As I understand it, bech32 has a quite low limit on its maximum size (90 bytes encoded right?), so it will be unsuitable for any of the larger things, such as most of the certificates. Indeed wouldn't it be unsuitable for extended keys too?

@dcoutts this is partially correct. Bech32 embedded **error-detection** works very well for chain smaller than 90 characters, but it is said to work _reasonably well_ up to 1024 characters. Above 89 characters, the probability of detecting errors becomes a bit smaller, yet remains okay.

**Comment by ilap** [comment]
I would prefer smaller HRPs like in the SLIP-173 @SebastienGllmt mentioned also, the "addr" common prefix is too general for the users especially when there would be native multi-asset tokens available for Cardano very soon.

Other reason is having less complex QR codes for the native currency and tokens.
Bech32 has 23 chars and 9 numbers, which can be used in HRP too, that would allow a very compact QR code for a hash28 token addresses as that mode can use only the 32-chars as a character set. As and example `ca1xxxxxxx` as `c` and `a` int the bech32 character set.

Also, we need to distinguish between the wallet- and the chain addresses (serialised keys, hash scripts etc on chain as an example). The wallet addresses are just for humans i.e. wallet users etc. for giving some meaning to or just wrap these chain-addresses.
But, as an example a user could choose his/her wallet address' prefix anyway(in a wallet that have that feature), as it won't affect the real on-chain addresses, meaning `addr1xxxx` or `this_is_my1xxxx` can refer to the exactly the same address on the chain.

[Anyway, see detailed QR examples for different character sets used with bech32]( https://bitcoinops.org/en/bech32-sending-support/#creating-more-efficient-qr-codes-with-bech32-addresses)

**Pull request merged**